### PR TITLE
Update: Updated all.sh

### PR DESCRIPTION
**Pull request by Pastucha**
1. Update: all.sh by ability to chose certain device we want to run.
2. Update: README.md by section how to run simulators filtered by name or API.
3. Sync the fork
4. Fixing small issues

### CODE REVIEW

**Comment by Pastucha** [comment]
@swistakm

File: all.sh [hunk 1]
```diff
@@ -1,10 +1,42 @@
 #!/usr/bin/env bash
 
+usage() {
+  echo "Usage: $0 [--help] [-k FILTER]"
+  echo "  --help          Display this help message"
```
Comment (swistakm) [inline all.sh]: there's no such option (i.e. `--help`), there's only `-h`:

```
$ ./all.sh --help
Invalid option: -
(...)
```

You can fix usage or figure out how to support short+long options.

File: all.sh [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
+      usage
+      ;;
+    :)
+      echo "Option -$OPTARG requires an argument." 1>&2
+      usage
+      ;;
+  esac
+done
+shift $((OPTIND -1))
+
```
Comment (swistakm) [inline all.sh:48]: A nice touch would be to check if we have any leftover non-option argument and return error if that's the case:

```
while [[ $# -gt 0 ]]; do
  echo "Too many arguments: ${@}"
  usage
done
```

**Review by swistakm — COMMENTED** [review]
Nice, there's only minor polishing needed.

Also, I've run this locally and noticed that if there's no match the script will enter a sleep loop however there's nothing running. It would be a good idea to set some flag that tells the script that at least one device simulator is running. If it is unset we should print error message and return non-zero exit code instead of entering the sleep loop.

**Comment by Pastucha** [comment]
@swistakm

**Comment by Pastucha** [comment]
@swistakm  done

**Pull request merged**